enhancements/oc/oc-login-external-oidc-integration.md

```diff
@@ -0,0 +1,228 @@
+  * `--auto-open-browser`: Specify browser is automatically opened or not. It is especially useful, if users want to use non-default browser.
+
+and OIDC specific flags in `oc login`;
+* `--exec-plugin`: Type indicates that which credentials exec plugin will be used by client-go. It only supports `oc-oidc` (uses `get-token` built-in command) currently, but can be easily extended.
+* `--issuer-url`: Issuer URL of the external OIDC issuer and mandatory.
+* `--client-id`: Client ID that is created in external OIDC issuer and mandatory.
+* `--client-secret`: Optional field that can be passed, if cluster administrators intentionally choose to distribute this.
+* `--extra-scopes`: Includes all the extra scopes in a comma separated format (e.g. `--extra-scopes=email,profile`) and optional.
+* `--callback-port`: Callback port where external OIDC issuer redirects to after flow is completed. If it is not specified, command picks a random unused port and concatenating it to 127.0.0.1 
+
+### Workflow Description
+
+oc login generates a stanza in kubeconfig supporting credentials exec plugin feature in client-go.
+
+For example;
+
+```shell
+$ oc login localhost:8443 ---issuer-url=https://oidc.issuer.url -exec-plugin=oc-oidc --client-id=client-id --extra-scopes=email,profile --callback-port=8080
+```
+
+results in;
+
+```yaml
+- name: localhost:8443
+  user:
+    exec:
+      apiVersion: client.authentication.k8s.io/v1
+      args:
+      - get-token
+      - --issuer-url=https://oidc.issuer.url
+      - --client-id=client-id
+      - --callback-address=127.0.0.1:8080
+      - --extra-scopes=email,profile
+      command: oc
+      env: null
+      installHint: Please be sure that oc is defined in $PATH to be executed as credentials
+        exec plugin
+      interactiveMode: IfAvailable
+      provideClusterInfo: false
+```
+
+Moreover, `oc login` command triggers authentication process by sending a request to Project API's whoami endpoint automatically so that
+client-go talks to `oc get-token` command to obtain id token (and also refresh token, if provider supports it).
+
+There are currently 7 authentication flows grouped around public and confidential clients. Due to the nature of `oc` that is being used widely on local, 
+we have decided to support authentication flows only for public clients for now but if there is any different use case in the future, this command can support confidential clients too. 
+
+Supportability (in terms of authentication flows) matrix of this feature is outlined below; 
+
+|           | Authorization Code<br/>(public/confidential)                        | Authorization Code + PKCE<br/>(public/confidential)                 | Implicit Flow<br/>(public) | Client Credentials<br/>(confidential)                                    | Refresh Token<br/>(public/confidential) | Device Code<br/>(public) | Password Grant<br/>(confidential) |
```

Comment:
If the "it just works for the console" row will vary based on public or confidential (I bet it will), let's separate the public/confidential into two differnet columns.

Reply:
Yes, it will vary based on public or confidential. Separated.